src/Modules/Data/Commands.Data/GetPowerBIDataset.cs

```diff
@@ -92,12 +92,12 @@ public override void ExecuteCmdlet()
                 this.WorkspaceId = this.Workspace.Id;
             }
 
-            if (this.ParameterSet.Equals(IdParameterSetName))
+            if (this.Id != default)
             {
                 this.Filter = $"id eq '{this.Id}'";
             }
 
-            if (this.ParameterSet.Equals(NameParameterSetName))
+            if (this.Name != default)
```

Comment:
Will the cmdlet fail if parameters from multiple parameter sets are provided? I'm assuming you're checking against default because these parameters are optional within the parameter set and the C# code sees them as the default value even when not provided.

Reply:
The parameter belongs to multiple parameter sets and doing it this way protects better against future changes. The parametersets are more to prevent this parameter from being populated when it shouldn't be used.